R/blob.R

```diff
@@ -19,48 +27,55 @@ setOldClass("blob")
 #'
 #' as.blob(c("Good morning", "Good evening"))
 blob <- function(...) {
-  new_blob(list(...))
+  x <- list2(...)
+  check_raw_list(x)
+  new_blob(x)
 }
 
-#' @export
-#' @rdname blob
-new_blob <- function(x) {
+check_raw_list <- function(x) {
+  quo <- enquo(x)
   if (!is_raw_list(x)) {
-    stop("`x` must be a list of raw vectors", call. = FALSE)
+    stop("`", as_label(quo), "` must be a list of raw vectors", call. = FALSE)
   }
-  structure(x, class = "blob")
 }
 
 #' @export
 #' @rdname blob
-as.blob <- function(x, ...) {
-  UseMethod("as.blob")
-}
-
-#' @export
-as.blob.blob <- function(x, ...) {
-  x
+new_blob <- function(x = list()) {
+  vec_assert(x, list())
+  new_list_of(x, ptype = raw(), class = "blob")
```

Comment:
If we're updating to the latest conventions, should be make this `blob_blob`, or is that likely to cause too many breakages in other packages?

Reply:
We could do both, or change the convention that if the package implements just one class, that name can be used as class name?

Reply:
That seems like a reasonable argument to leave as is.